src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -384,3 +353,102 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 
 	return FinalizePlan(planContext->plan, distributedPlan);
 }
+
+
+/*
+ * ShardPlacementForFunctionColocatedWithDistTable decides on a placement
+ * for delegating a procedure call that accesses a distributed table.
+ */
+static ShardPlacement *
+ShardPlacementForFunctionColocatedWithDistTable(DistObjectCacheEntry *procedure,
+												FuncExpr *funcExpr,
+												Var *partitionColumn,
+												CitusTableCacheEntry *cacheEntry,
+												PlannedStmt *plan)
+{
+	if (procedure->distributionArgIndex < 0 ||
```

Comment:
If the `distributionArgIndex` is not provided, then it is set to `NULL`, and consequently `procedure->distributionArgIndex` is set to 0.

If the value is not in these interval, the error makes sense. However I am not sure how this can happen. I can only think of breaking `pg_distobject` records manually